src/main/java/com/switchfully/parkshark_2024_10/division/DivisionService.java

```diff
@@ -0,0 +1,24 @@
+package com.switchfully.parkshark_2024_10.division;
+
+import com.switchfully.parkshark_2024_10.division.dto.CreateDivisionDto;
+import com.switchfully.parkshark_2024_10.division.dto.DivisionDto;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+@Service
+@Transactional
+public class DivisionService {
+    DivisionRepository divisionRepository;
+    DivisionMapper divisionMapper = new DivisionMapper();
+
+    public DivisionService(DivisionRepository divisionRepository) {
+        this.divisionRepository = divisionRepository;
+    }
+
+    public DivisionDto addDivision(CreateDivisionDto createDivisionDto){
+        Division division = divisionMapper.mapToDivision(createDivisionDto);
```

Comment:
We could add the "map `director_id` to a Director" here, or in the `mapToDivision` method itself.

As the `divisionMapper` is not a Bean (and should probably not be Bean) we should call a service here that takes in a `director_id` and checks if the Director exists (and other checks if needed). Then it would get a Director back which you can pass through to the `mapToDivision` method.